src/Resolvers/Microsoft.NET.Sdk.WorkloadManifestReader/SdkDirectoryWorkloadManifestProvider.cs

```diff
@@ -273,13 +297,22 @@ void ProbeDirectory(string manifestDirectory, string featureBand)
             //  Assume that if there are any versioned subfolders, they are higher manifest versions than a workload manifest directly in the specified folder, if it exists
             if (manifestVersionDirectories.Any())
             {
-                return (manifestId, manifestVersionDirectories.First().directory);
+                return (manifestId, manifestVersionDirectories.First().directory, manifestVersionDirectories.First().version);
             }
             else if (File.Exists(Path.Combine(manifestDirectory, "WorkloadManifest.json")))
             {
-                return (manifestId, manifestDirectory);
+                var manifestPath = Path.Combine(manifestDirectory, "WorkloadManifest.json");
+                try
+                {
+                    var manifestContents = WorkloadManifestReader.ReadWorkloadManifest(manifestId, File.OpenRead(manifestPath), manifestPath);
+                    return (manifestId, manifestDirectory, new ReleaseVersion(manifestContents.Version));
+                }
+                catch
+                {
+                    return (manifestId, manifestDirectory, null);
+                }
```

Comment:
nit: Makes it more obvious that the `catch` doesn't do anything.
```suggestion
                ReleaseVersion releaseVersion = null;
                try
                {
                    var manifestContents = WorkloadManifestReader.ReadWorkloadManifest(manifestId, File.OpenRead(manifestPath), manifestPath);
                    releaseVersion = new ReleaseVersion(manifestContents.Version);
                }
                catch { }

                return (manifestId, manifestDirectory, releaseVersion);
```

Reply:
This is an interesting suggestion...on the one hand, it's longer and less attractive to me. On the other hand, you're right that it makes it more clear that the catch block doesn't do anything. I think clarity is more important, and it makes it easy for me to add a comment in the catch block to explain this.

Thanks for the suggestion!